idea/idea-analysis/src/org/jetbrains/kotlin/idea/caches/KotlinShortNamesCache.kt

```diff
@@ -85,30 +87,78 @@ class KotlinShortNamesCache(private val project: Project) : PsiShortNamesCache()
         dest.addAll(allClassNames)
     }
 
-    override fun getMethodsByName(name: String, scope: GlobalSearchScope): Array<PsiMethod>
-            = emptyArray()
+    override fun getMethodsByName(name: String, scope: GlobalSearchScope): Array<PsiMethod> {
+        val propertiesIndex = KotlinPropertyShortNameIndex.getInstance()
+        val functionIndex = KotlinFunctionShortNameIndex.getInstance()
 
-    override fun getMethodsByNameIfNotMoreThan(name: String, scope: GlobalSearchScope, maxCount: Int): Array<PsiMethod>
-            = emptyArray()
+        val kotlinFunctionsPsi = functionIndex.get(name, project, scope)
+                .flatMap { LightClassUtil.getLightClassMethods(it) }
 
-    override fun getFieldsByNameIfNotMoreThan(s: String, scope: GlobalSearchScope, i: Int): Array<PsiField>
-            = emptyArray()
+        val propertyAccessorsPsi = getPropertyNamesCandidatesByAccessorName(Name.identifier(name))
+                .flatMap { propertiesIndex.get(it.asString(), project, scope) }
+                .flatMap { LightClassUtil.getLightClassPropertyMethods(it).allDeclarations }
+                .filter { it.name == name }
+                .map { it as? PsiMethod }
+
+        return (propertyAccessorsPsi + kotlinFunctionsPsi).filterNotNull().toList().toTypedArray()
+    }
+
+    override fun getMethodsByNameIfNotMoreThan(name: String, scope: GlobalSearchScope, maxCount: Int): Array<PsiMethod> {
+        val methods = getMethodsByName(name, scope)
+        if (methods.size > maxCount)
+            return PsiMethod.EMPTY_ARRAY
+        return methods
+    }
+
+    override fun getFieldsByNameIfNotMoreThan(name: String, scope: GlobalSearchScope, maxCount: Int): Array<PsiField> {
+        val fields = getFieldsByName(name, scope)
+        if (fields.size > maxCount)
+            return PsiField.EMPTY_ARRAY
+        return fields
+    }
 
     override fun processMethodsWithName(name: String, scope: GlobalSearchScope, processor: Processor<PsiMethod>): Boolean
             = ContainerUtil.process(getMethodsByName(name, scope), processor)
 
-    override fun getAllMethodNames(): Array<String>
-            = emptyArray()
+    override fun getAllMethodNames(): Array<String> {
+        val functionIndex = KotlinFunctionShortNameIndex.getInstance()
+        val functionNames = functionIndex.getAllKeys(project)
+
+        val propertiesIndex = KotlinPropertyShortNameIndex.getInstance()
+        val globalScope = GlobalSearchScope.allScope(project)
+        val propertyAccessorNames = propertiesIndex.getAllKeys(project)
+                .flatMap { propertiesIndex.get(it, project, globalScope) }
```

Comment:
This will be extremely slow. It's OK to return names for which `getMethodsByName()` will then not find any methods, but it's not OK to look up every single method in the index.